syntax = "proto3";

package posts;

message GetCountRequest {
  string targetType = 1;
  int32 post_id = 2;
}

message GetCountResponse { int32 counter = 1; }

message DayData {
  string date = 1;
  int32 value = 2;
}

message GetDynamicsRequest {
  string targetType = 1;
  int32 post_id = 2;
}

message GetDynamicsResponse { repeated DayData dayData = 1; }

message GetTopPostsRequest {
  string targetType = 1;
  int32 limit = 2;
}

message GetTopUsersRequest {
  string targetType = 1;
  int32 limit = 2;
}

message TopStatsResponse {
  int32 id = 1;
  int32 value = 2;
}

message GetTopResponse { repeated TopStatsResponse items = 1; }

message GetPostStatsRequest { int32 post_id = 1; }

message GetPostStatsResponse {
  int32 views = 1;
  int32 comments = 2;
  int32 likes = 3;
}

service StatsService {
  rpc GetCount(GetCountRequest) returns (GetCountResponse);
  rpc GetDynamics(GetDynamicsRequest) returns (GetDynamicsResponse);
  rpc GetTopPosts(GetTopPostsRequest) returns (GetTopResponse);
  rpc GetTopUsers(GetTopUsersRequest) returns (GetTopResponse);
  rpc GetPostStats(GetPostStatsRequest) returns (GetPostStatsResponse);
}